src/Components/Patient/SampleViewAdmin.tsx

```diff
@@ -41,69 +37,34 @@ export default function SampleViewAdmin() {
     limit: 10,
     cacheBlacklist: ["patient_name", "district_name"],
   });
-  const dispatch: any = useDispatch();
-  const initialData: any[] = [];
   let manageSamples: any = null;
-  const [sample, setSample] = useState<Array<SampleTestModel>>(initialData);
-  const [isLoading, setIsLoading] = useState(false);
-  const [totalCount, setTotalCount] = useState(0);
   const [fetchFlag, callFetchData] = useState(false);
-  const [facilityName, setFacilityName] = useState("");
   const [statusDialog, setStatusDialog] = useState<{
     show: boolean;
     sample: SampleTestModel;
   }>({ show: false, sample: {} });
 
-  useEffect(() => {
-    async function fetchData() {
-      if (!qParams.facility) return setFacilityName("");
-      const res = await dispatch(getAnyFacility(qParams.facility));
-      setFacilityName(res?.data?.name);
-    }
-
-    fetchData();
-  }, [dispatch, qParams.facility]);
-
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-      const res = await dispatch(
-        getTestList({
-          limit: resultsPerPage,
-          offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
-          patient_name: qParams.patient_name || undefined,
-          district_name: qParams.district_name || undefined,
-          status: qParams.status || undefined,
-          result: qParams.result || undefined,
-          facility: qParams.facility || "",
-          sample_type: qParams.sample_type || undefined,
-        })
-      );
-      if (!status.aborted) {
-        if (res && res.data) {
-          setSample(res.data.results);
-          setTotalCount(res.data.count);
-        }
-        setIsLoading(false);
-      }
+  const { data: facilityData } = useQuery(routes.getAnyFacility, {
+    pathParams: {
+      id: qParams.facility,
     },
-    [
-      dispatch,
-      qParams.page,
-      qParams.patient_name,
-      qParams.district_name,
-      qParams.status,
-      qParams.result,
-      qParams.facility,
-      qParams.sample_type,
-    ]
-  );
+    prefetch: !!qParams.facility,
+  });
 
-  useAbortableEffect(
-    (status: statusType) => {
-      fetchData(status);
-    },
-    [fetchData, fetchFlag]
+  const { loading: isLoading, data: sampeleData } = useQuery(
+    routes.getTestSampleList,
+    {
+      query: {
+        limit: resultsPerPage,
+        offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
+        patient_name: qParams.patient_name || undefined,
+        district_name: qParams.district_name || undefined,
+        status: qParams.status || undefined,
+        result: qParams.result || undefined,
+        facility: qParams.facility || "",
```

Comment:
These are just filters. Why `undefined` in some places and why `""` empty string in some places?

```suggestion
        facility: qParams.facility || undefined,
```